### PR TITLE
Disable managed_ptr stuff in the HIP build

### PR DESCRIPTION
**Pull request by adayton1**
Virtual functions are not support in HIP device calls, so managed_ptr must be disabled.

### CODE REVIEW

**Review by robinson96 — APPROVED** [review]
LGTM

File: src/care/host_device_ptr.h [hunk 1]
```diff
@@ -438,6 +438,7 @@ namespace care {
    void using_host_device_ptr_outside_of_raja_loop_not_allowed(T foo); 
 } // namespace care
 
+#if defined(CARE_ENABLE_MANAGED_PTR)
```
Comment (dtaller) [inline src/care/host_device_ptr.h:441]: Sorry, I may be missing something, but why are you putting this here? We are declaring getRawPointers with

care::host_device_ptr<T>

not managed_ptr , so it should be a different case?

Comment (adayton1) [reply src/care/host_device_ptr.h:441]: The original getRawPointers functions are defined in managed_ptr.hpp. So if managed_ptr support is disabled in CARE, there is no reason for this function.

**Comment by dtaller** [comment]
> Virtual functions are not support in HIP device calls, so managed_ptr must be disabled.


Do we know if that will be changed/fixed for future versions of HIP? We should ask Kostas

**Review by dtaller — APPROVED** [review]
Looks good, just ask Kostas if virtual functions will be ok on the device in the future

**Pull request merged**